projects/python-client/src/datapane/client/api/report/core.py

```diff
@@ -166,6 +168,14 @@ def save(
 
         save_report(self, path, open, standalone, name, author, formatting, cdn_base)
 
+    @staticmethod
+    def from_notebook(opt_out: bool = True) -> App:
```

Comment:
I think we can inline `notebook_to_app` here directly, it's not doing much itself and unsure if will be used standalone. The recursive nested imports both here and there suggest that we can probably simplify things a bit

Reply:
Agree on the recursive nested imports!

I kept it this way because it's about to grow with 1) the calls to the notebook vs cache check, and 2) figuring out which template to use. I'm sure it will continue to grow as we add new features. I thought it would be nice to keep it all within our `ipython_utils`, but if you prefer, I can move it to `from_notebook`.

Reply:
Happy to keep as an external helper method - I think let's prepend with a `_` then to mark it for internal use right now (we don't want to mark too much in the public API as then it makes it hard to modify going forwards).

I think we should remove the recursive import - the `App.from_notebook` classmethod should be responsible for constructing the `App` itself, and then this function can just return the blocks for use within the classmethod.

Reply:
No problem! As we don't want too much in the public API, it makes sense to get rid of it. I've done so and moved it into `App.from_notebook`.